script.js

```diff
@@ -1,12 +1,185 @@
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+let allShows = getAllShows().sort(function (a, b) {
+  if (a.name < b.name) {
+    return -1;
+  } else if (a.name > b.name) {
+    return 1;
+  } else {
+    return 0;
+  }
+});
+let allEpisodes;
+const body = document.querySelector("body");
+let selectedShowsId = allShows.map((p) => p.id);
+let defaulShowId = "82";
+
+function getFetchForEpisodes(showsId) {
+  fetch(`https://api.tvmaze.com/shows/${showsId}/episodes`)
+    .then(function (response) {
+      return response.json();
+    })
+    .then(function (data) {
+      allEpisodes = data;
+      makePageForEpisodes(allEpisodes);
+      makeEpisodeList(allEpisodes);
+    })
+    .catch((err) => console.log(err));
+}
+
+//header
+const header = document.createElement("header");
+header.id = "header";
+header.innerHTML = "<h1>TV SHOW</h1>";
+body.appendChild(header);
+
+//navbar
+const navBar = document.createElement("nav");
+navBar.id = "nav";
+body.appendChild(navBar);
+
+//main container
+let divElement = document.createElement("div");
+divElement.id = "main";
+body.appendChild(divElement);
+
+//select shows
+let selectedShows = document.createElement("select");
+selectedShows.id = "shows-list";
+const firstShowOption = document.createElement("option");
+selectedShows.prepend(firstShowOption);
+firstShowOption.innerText = "< SELECT SHOWS >";
+selectedShows.id = "select_shows";
+navBar.prepend(selectedShows);
+selectedShows.addEventListener("change", getShowId);
+
+function makeShowsList(showsList) {
+  showsList.forEach((show) => createShowsList(show));
+}
+
+function createShowsList(newShows) {
+  let showsOption = document.createElement("option");
+  selectedShows.appendChild(showsOption);
+  showsOption.value = newShows.id;
+  showsOption.innerHTML = newShows.name;
+}
+
+function getShowId() {
+  selectedEpisode.innerHTML = "";
+  let findShowsId = allShows.find((shows) => shows.id == selectedShows.value);
+  if (findShowsId == null) {
+    getFetchForEpisodes(defaulShowId);
+  } else {
+    getFetchForEpisodes(findShowsId.id);
+  }
+}
+
+// select episodes
+let selectedEpisode = document.createElement("select");
+let firstOption = document.createElement("Option");
+selectedEpisode.prepend(firstOption);
+firstOption.innerText = "< SELECT EPISODES >";
+selectedEpisode.id = "select_episode";
+navBar.appendChild(selectedEpisode);
+selectedEpisode.addEventListener("change", selectEpisode);
+
+function makeEpisodeList(newEpisodes) {
+  newEpisodes.forEach((episode) => createEpisodesList(episode));
+}
+
+function createEpisodesList(episode) {
```

Comment:
```suggestion
function createEpisodeOption(episode) {
```

Reply:
This function does the work for creating a single episode so avoid giving it a plural name - just a small refactor there